hydra-node/src/Hydra/Chain/Direct/Wallet.hs

```diff
@@ -146,11 +147,12 @@ newTinyWallet tracer networkId (vk, sk) queryWalletInfo queryEpochInfo = do
       { getUTxO
       , getSeedInput = fmap (fromLedgerTxIn . fst) . findFuelUTxO <$> getUTxO
       , sign = signWith sk
-      , coverFee = \lookupUTxO partialTx -> do
+      , coverFee = \lookupUTxO partialTx' -> do
           -- XXX: We should query pparams here. If not, we likely will have
           -- wrong fee estimation should they change in between.
           epochInfo <- queryEpochInfo
           WalletInfoOnChain{walletUTxO, pparams, systemStart} <- readTVarIO walletInfoVar
+          let partialTx = unsafeBuildTransaction $ setProtocolParams pparams partialTx'
```

Comment:
The point of this change is to not need to use `unsafeBuildTransaction`. So do use `createAndValidateTransactionBody` here.